From d30f8a09fc297b887d36cb077800a2a2a97dc9ff Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 30 Jan 2022 01:14:30 +0200
Subject: [PATCH]  [SW-65638] gaudi2: release: load local itb

Simplify the loading of a local itb file directly to the device
via the use of the host's FWL.

Change-Id: I68b2362b84d1d0a90da9043e21dabccc43771f53
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index a369010..90b67c2 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -670,6 +670,33 @@
 	return 0
 }
 
+function remote_fwl_loacl_itb()
+{
+	local local_itb_path=$1
+	local file_name=$(basename ${local_itb_path})
+
+	shift
+
+	 [ ! -f ${local_itb_path:-""} ] && {
+		 perr "fwl-loc: File not exist: ${local_itb_path}"
+		 return 1
+	}
+
+	#TODO move the host_* from SA/HLS modules
+	#     into a common module, so calling host_cp_to etc.
+	#     will be legit.
+
+	host_cp_to ${local_itb_path} /tmp/${file_name}
+
+	remote_fwl -f /tmp/${file_name} *$ >&3
+	[ $? -ne 0 ] && {
+		perr "[${pci_dev}] Failed to update eeprom: $file_name"
+		return 1
+	} || {
+		pok "[${pci_dev}] Succeed to update eeprom: $file_name"
+	}
+}
+
 function read_eeprom()
 {
 	local oam=$1
@@ -1105,6 +1132,23 @@
 
 	#redirect remote_fwl ${fwl_args[@]}
 	remote_fwl ${fwl_args[@]} >&3
+}
+
+function remote_fwl_local_itb_op()
+{
+	local loc_file
+
+	[ ${fwl_loc_f:-0} -eq 0 ] && retrun 0
+	[ ${#fwl_loc_args[@]} -eq 0 ] && return 1
+
+	loc_file=${fwl_loc_itb_args[0]}
+	unset fwl_loc_itb_args[0]
+
+	[ ! -f ${loc_file} -eq 0 ] && {
+		perr "fwl-loc: local file not exist: ${loc_file}"
+		return 1
+	}
+
 }
 
 function handle_ignore_op()
@@ -1932,6 +1976,7 @@
 	mods_var_register config_link show_env
 	mods_var_register copy_files
 	mods_var_register fwl
+	mods_var_register fwl_loc
 
 	# User config files and platform selection was already
 	# handled in the early param handling
